core/platform.h

```diff
@@ -90,6 +90,11 @@ typedef _microstep_t microstep_t;
 
 #ifdef NUMBER_OF_WORKERS
 
+/**
+ * @brief Get the number of cores on the host machine.
+ */
+extern int lf_host_cores();
```

Comment:
Maybe a more descriptive name would be `num_of_cores` or `available_cores`? I don't think "host" is very meaningful; this being part of the platform API makes it obvious that this pertains to the platform.

Reply:
We don't have fully qualified names in C, so you would just see `available_cores()` in the code with no context where it is defined, unless you look for it. I think `lf_host_cores` is descriptive enough (maybe `lf_host_num_cores`?), but if you think `available_cores` is better, we could use that too.

Reply:
What else would the number of reported available cores refer to but the number of available core on which the code executes? The number of available cores on a different machine? Who would assume that?

Reply:
Sure, we can use `available_cores()`. That's fine with me.

Reply:
With the `lf_` prefix though, right?